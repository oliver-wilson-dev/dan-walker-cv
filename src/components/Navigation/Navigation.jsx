import React from 'react';
import { Link } from 'react-router-dom';

import styles from './Navigation.module.css';

const Navigation = () => (
  <nav className={styles.navigation}>
    <ul className={styles.navItems}>
      <li className={styles.navItem}>
        <Link className={styles.link} to="/">Home</Link>
      </li>
      <li className={styles.navItem}>
        <Link className={styles.link} to="/about">About</Link>
      </li>
      <li className={styles.navItem}>
        <Link className={styles.link} to="/contact">Contact</Link>
      </li>
    </ul>
  </nav>
);

export default Navigation;
