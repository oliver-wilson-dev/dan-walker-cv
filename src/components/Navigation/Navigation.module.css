@import "../../styles/globalVars.css";

.navigation {
  height: $navigation-height;
  position: fixed;
  top: 0;
  background-color: #f8f8f8;
  width: 100%;
  padding: 0px;
  margin: 0px;
  box-shadow: 0px 0px 8px #000000;
  display: flex;
  justify-content: center;
  align-items: center;
}

.navItems {
  padding: 0px;
  margin: 0px;
  list-style-type: none;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-grow: 1;
  cursor: pointer;
  max-width: $max-content-width;
  height: 100%;
}

.navItem {
  display: flex;
  height: 100%;
}

.link {
  padding: 15px 30px;
  padding: 15px;
  text-decoration: none;
  color: #000000;

  &:hover {
    background-color: #fcbf49;
  }
}

@media (min-width: 768px) {
  .link {
    padding: 15px 30px;
  }

  .navItems {
    justify-content: flex-end;
  }
}
