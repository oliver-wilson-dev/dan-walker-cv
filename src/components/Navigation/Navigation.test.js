import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

jest.mock('react-router-dom', () => ({
  Link: ({ children }) => <>{children}</>
}));

const render = () => shallow(<Navigation />);

it('renders correctly', () => {
  expect(render()).toMatchSnapshot();
});
