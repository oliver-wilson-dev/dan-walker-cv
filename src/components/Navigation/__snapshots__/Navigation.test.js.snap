// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navigation"
>
  <ul
    className="navItems"
  >
    <li
      className="navItem"
    >
      <Link
        className="link"
        to="/"
      >
        Home
      </Link>
    </li>
    <li
      className="navItem"
    >
      <Link
        className="link"
        to="/about"
      >
        About
      </Link>
    </li>
    <li
      className="navItem"
    >
      <Link
        className="link"
        to="/contact"
      >
        Contact
      </Link>
    </li>
  </ul>
</nav>
`;
